external/anomaly/anomaly_classification/configs/padim/configuration.yaml

```diff
@@ -49,65 +38,45 @@ dataset:
       type: UI_RULES
     value: 5
     visible_in_ui: true
-    warning:
-      Increasing this value may cause the system to use more memory than available,
+    warning: Increasing this value may cause the system to use more memory than available,
       potentially causing out of memory errors, please update with caution.
-  num_workers:
-    affects_outcome_of: NONE
-    default_value: 0
-    description:
-      Increasing this value might improve training speed however it might
-      cause out of memory errors. If the number of workers is set to zero, data loading
-      will happen in the main training thread.
-    editable: true
-    header: Number of cpu threads to use during batch generation
-    max_value: 8
-    min_value: 0
-    type: INTEGER
-    ui_rules:
-      action: DISABLE_EDITING
-      operator: AND
-      rules: []
-      type: UI_RULES
-    value: 0
-    visible_in_ui: true
-    warning: null
   type: PARAMETER_GROUP
   visible_in_ui: true
-
-trainer:
-  description: Parameters for the PyTorch Lightning Trainer
-  header: Trainer Parameters
-  type: PARAMETER_GROUP
-  visible_in_ui: false
-  num_sanity_val_steps:
+description: Configuration for Padim
+header: Configuration for Padim
+id: ''
+model:
+  description: Model Parameters
+  header: Model Parameters
+  name:
     affects_outcome_of: NONE
-    default_value: -1
-    description:
-      This is needed to ensure that the pytorch lightning trainer starts a
-      validation run on the full validation set before training.
+    default_value: padim
+    description: Name of the model that should be used for training.
     editable: false
-    header: Number of validation steps in the sanity check
-    max_value: -1
-    min_value: -1
-    type: INTEGER
+    enum_name: ModelName
+    header: Model Name
+    options:
+      PADIM: padim
+      STFPM: stfpm
```

Comment:
should stfpm be in padim configuration?

Reply:
Another question is whether we need model name in the configuration at all? Seems like this is planned to be addressed with the model templates, not from configuration

Reply:
The `configuration.yaml` files are generated automatically by parsing the python class, so it also automatically adds all options for the selectables (in this case padim and stfpm).

Reply:
> Another question is whether we need model name in the configuration at all? Seems like this is planned to be addressed with the model templates, not from configuration

I think any parameters that we assign in `template.yaml` should also be in `configuration.yaml` or it will give an error.

Reply:
If `configuration.yaml` is generated then should we add it to `.gitignore`?

Reply:
Or is the plan to generate from our side each time we make some changes?

Reply:
OTE needs those files in order to instantiate a task, so they should be in the repo. For now the plan is to generate them using a pre-commit hook.

Reply:
> I think any parameters that we assign in template.yaml should also be in configuration.yaml or it will give an error.

Are we assigning those values in template.yaml file? I think model is selected from [here](https://github.com/openvinotoolkit/training_extensions/blob/dick/anomalib-config-classes/external/anomaly/anomaly_classification/configs/padim/template.yaml#L2)

Reply:
That is why ote-reviewers ask us to remove model names from configuration.yaml file